### PR TITLE
chore(log): fix the log level and use key values

### PR DESCRIPTION
**Pull request by padlar**
Previous attempt to dump the logs didnt work as expected. Logr levels and zap level didnt really match, this should fix the logr level and use the args as key values.

Verified the logs in the output 

```

  Timeline >>
  2023-01-26T10:02:42+01:00     INFO    Reconciling     {"controller": "containerimagescan", "controllerGroup": "stas.statnett.no", "controllerKind": "ContainerImageScan", "ContainerImageScan": {"name":"application-example-app-a309a","namespace":"default"}, "namespace": "default", "name": "application-example-app-a309a", "reconcileID": "7343d6d1-a
eea-4979-818e-2b8df6ebe3a0"}
  2023-01-26T10:02:42+01:00     INFO    Reconciling     {"controller": "job", "controllerGroup": "batch", "controllerKind": "Job", "Job": {"name":"application-example-app-a309a-ab8ae","namespace":"image-scanner-jobs"}, "namespace": "image-scanner-jobs", "name": "application-example-app-a309a-ab8ae", "reconcileID": "5d4de186-9568-4875-8317-7ad5c6ea
4c76"}
  2023-01-26T10:02:42+01:00     INFO    Reconciling job with status.    {"controller": "job", "controllerGroup": "batch", "controllerKind": "Job", "Job": {"name":"application-example-app-a309a-ab8ae","namespace":"image-scanner-jobs"}, "namespace": "image-scanner-jobs", "name": "application-example-app-a309a-ab8ae", "reconcileID": "5d4de186-9568-48
75-8317-7ad5c6ea4c76", "status": {"succeeded":1}}
  2023-01-26T10:02:42+01:00     INFO    Patching CIS status with success job status     {"controller": "job", "controllerGroup": "batch", "controllerKind": "Job", "Job": {"name":"application-example-app-a309a-ab8ae","namespace":"image-scanner-jobs"}, "namespace": "image-scanner-jobs", "name": "application-example-app-a309a-ab8ae", "reconcileID": "
5d4de186-9568-4875-8317-7ad5c6ea4c76"}
  2023-01-26T10:02:44+01:00     INFO    Patched CIS status      {"controller": "job", "controllerGroup": "batch", "controllerKind": "Job", "Job": {"name":"application-example-app-a309a-ab8ae","namespace":"image-scanner-jobs"}, "namespace": "image-scanner-jobs", "name": "application-example-app-a309a-ab8ae", "reconcileID": "5d4de186-9568-4875-8317-
7ad5c6ea4c76", "reason": "VulnerabilityOverflow", "error": null}
  [FAILED] in [It] - /home/sunils/sandbox/padlar-iso/internal/controller/stas/scan_job_controller_test.go:103 @ 01/26/23 10:02:44.601
  << Timeline

  [FAILED] Expected
      <string>: Stalled
  to equal
      <string>: Stalled2
  In [It] at: /home/sunils/sandbox/padlar-iso/internal/controller/stas/scan_job_controller_test.go:103 @ 01/26/23 10:02:44.601
```

### CODE REVIEW

**Review by mikaelol — DISMISSED** [review]

File: internal/controller/stas/scan_job_controller.go [hunk 1]
```diff
@@ -171,7 +171,7 @@ func (r *ScanJobReconciler) reconcile() reconcile.Func {
 func (r *ScanJobReconciler) reconcileJob(ctx context.Context, job *batchv1.Job) error {
 	cisList := &stasv1alpha1.ContainerImageScanList{}
 
-	logf.FromContext(ctx).V(3).Info("Reconciling job %q with status %+v", job.Name, job.Status)
+	logf.FromContext(ctx).V(-3).Info("Reconciling job with status.", "job", job.Name, "status", job.Status)
```
Comment (erikgb) [inline internal/controller/stas/scan_job_controller.go:174]:
```suggestion
	logf.FromContext(ctx).V(-3).Info("Reconciling job with status.", "job", job.Name, "jobStatus", job.Status)
```

File: internal/controller/stas/scan_job_controller.go [hunk 2]
```diff
@@ -201,7 +201,7 @@ func (r *ScanJobReconciler) reconcileJob(ctx context.Context, job *batchv1.Job)
 	if err != nil {
 		switch {
 		case staserrors.IsJobPodNotFound(err), staserrors.IsScanJobContainerWaiting(err):
-			logf.FromContext(ctx).V(3).Info("Failed to get job %q logs %s", job.Name, err)
+			logf.FromContext(ctx).V(-3).Info("Failed to get logs.", "job name", job.Name, "error", err)
```
Comment (erikgb) [inline internal/controller/stas/scan_job_controller.go:204]:
```suggestion
			logf.FromContext(ctx).V(-3).Info("Failed to get logs.", "job", job.Name, "error", err)
```

**Review by erikgb — CHANGES_REQUESTED** [review]
I think we should be consistent when logging.

File: internal/controller/stas/scan_job_controller.go [hunk 3]
```diff
@@ -171,7 +171,7 @@ func (r *ScanJobReconciler) reconcile() reconcile.Func {
 func (r *ScanJobReconciler) reconcileJob(ctx context.Context, job *batchv1.Job) error {
 	cisList := &stasv1alpha1.ContainerImageScanList{}
 
-	logf.FromContext(ctx).V(3).Info("Reconciling job %q with status %+v", job.Name, job.Status)
+	logf.FromContext(ctx).V(-3).Info("Reconciling job with status.", "status", job.Status)
```
Comment (erikgb) [inline internal/controller/stas/scan_job_controller.go:174]:
```suggestion
	logf.FromContext(ctx).V(-3).Info("Reconciling", "status", job.Status)
```

File: internal/controller/stas/scan_job_controller.go [hunk 4]
```diff
@@ -201,7 +201,7 @@ func (r *ScanJobReconciler) reconcileJob(ctx context.Context, job *batchv1.Job)
 	if err != nil {
 		switch {
 		case staserrors.IsJobPodNotFound(err), staserrors.IsScanJobContainerWaiting(err):
-			logf.FromContext(ctx).V(3).Info("Failed to get job %q logs %s", job.Name, err)
+			logf.FromContext(ctx).V(-3).Info("Failed to get job logs.", "error", err)
```
Comment (erikgb) [inline internal/controller/stas/scan_job_controller.go:204]:
```suggestion
			logf.FromContext(ctx).V(-3).Error(err, "While getting logs")
```

File: internal/controller/stas/scan_job_controller.go [hunk 5]
```diff
@@ -216,10 +216,10 @@ func (r *ScanJobReconciler) reconcileJob(ctx context.Context, job *batchv1.Job)
 	}(logs)
 
 	if job.Status.Succeeded > 0 {
-		logf.FromContext(ctx).V(3).Info("Patching CIS %q status with success job status", cis.Name)
+		logf.FromContext(ctx).V(-3).Info("Patching CIS status with success job status")
```
Comment (erikgb) [inline internal/controller/stas/scan_job_controller.go:219]:
```suggestion
		logf.FromContext(ctx).V(-3).Info("Patching CIS status", "status", "Succeeded")
```

File: internal/controller/stas/scan_job_controller.go [hunk 6]
```diff
@@ -216,10 +216,10 @@ func (r *ScanJobReconciler) reconcileJob(ctx context.Context, job *batchv1.Job)
...
 		return r.reconcileCompleteJob(ctx, job.Name, logs, cis)
 	} else {
-		logf.FromContext(ctx).V(3).Info("Patching CIS %q status with failed job status", cis.Name)
+		logf.FromContext(ctx).V(-3).Info("Patching CIS status with failed job status")
```
Comment (erikgb) [inline internal/controller/stas/scan_job_controller.go:222]:
```suggestion
		logf.FromContext(ctx).V(-3).Info("Patching CIS status", "status", "Failed")
```

**Review by erikgb — CHANGES_REQUESTED** [review]

File: internal/controller/stas/scan_job_controller.go (hunk 4 above)
Comment (padlar) [reply internal/controller/stas/scan_job_controller.go:204]: i think it should be ok

Comment (padlar) [reply internal/controller/stas/scan_job_controller.go]: Error dumps stacktrace which is probably not required

```
  2023-01-26T11:16:01+01:00     ERROR   While getting logs      {"controller": "job", "controllerGroup": "batch", "controllerKind": "Job", "Job": {"name":"application-example-app-a309a-ab8ae","namespace":"image-scanner-jobs"}, "namespace": "image-scanner-jobs", "name": "application-example-app-a309a-ab8ae", "reconcileID": "9afb873e-467d-420d-b865-
8cf6d0e41539", "error": "test error"}
  github.com/statnett/image-scanner-operator/internal/controller/stas.(*ScanJobReconciler).reconcileJob
        /home/sunils/sandbox/padlar-iso/internal/controller/stas/scan_job_controller.go:220
  github.com/statnett/image-scanner-operator/internal/controller/stas.(*ScanJobReconciler).reconcile.func1.1
        /home/sunils/sandbox/padlar-iso/internal/controller/stas/scan_job_controller.go:164
  github.com/statnett/image-scanner-operator/internal/controller.Reconcile
        /home/sunils/sandbox/padlar-iso/internal/controller/reconcile.go:16
  github.com/statnett/image-scanner-operator/internal/controller/stas.(*ScanJobReconciler).reconcile.func1
        /home/sunils/sandbox/padlar-iso/internal/controller/stas/scan_job_controller.go:167
  sigs.k8s.io/controller-runtime/pkg/reconcile.Func.Reconcile
        /home/sunils/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/reconcile/reconcile.go:102
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
        /home/sunils/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:122
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /home/sunils/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:323
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /home/sunils/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:274
  sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /home/sunils/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:235
```

will revert this back to info

**Review by erikgb — APPROVED** [review]
LGTM!

**Pull request merged**